### DISCUSSION TITLE
Berry virtual relay

### DISCUSSION THREAD

**Discussion by speccy88** [arendst/Tasmota#19470] [Ideas]
Would it be possible to have an "easy" way to add a virtual relay to tasmota that would be controlled via hooks/callback with berry to update/query the state and have all the MQTT to work out of the box like that others relay but that would not be linked to an IO pin in the configuration. That way, it would be easy to add relays that behave like others but are controlled via http or serial or any other way.

I am aware of the documentation but a lot of things has to be done manually
https://tasmota.github.io/docs/Berry-Cookbook/#adding-a-button-to-the-main-menu

**Comment by barbudor** [comment]
You can easily add virtual relays in Berry
Actually you can have a look at the Berry sample HC595 driver that creates relays

**Comment by sfromis** [comment]
Virtual relays are by setting [tasmota.global.devices_present](https://tasmota.github.io/docs/Berry/#low-level-access-to-tasmota-globals-and-settings)
